### ISSUE TITLE
Simplify javascript client wire up to http

### ISSUE THREAD

**Issue by davidfowl** [aspnet/SignalR#838]
https://github.com/davidfowl/SignalRCoreAlphaSample/blob/350420c3b075c02e2ae31431e0f698072fcfe721/Sample/wwwroot/index.html#L9-L10

I think we're going to have to support defaulting the IConnection impl to HttpConnection in the javascript client.

Basically this:

https://github.com/aspnet/SignalR/blob/eec4b33cff09382285766844b8054d3e65fa1131/client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts#L31

We accept string url or IConnection.

```typescript
 constructor(connection: IConnection | string, options: IHubConnectionOptions = {}) { 
```

On top of that, we should support relative paths that will default to document.blah (if available, we can just detect if there's a document and if not, throw for the nodejs scenario). Basically making it so we don't have to do this:

https://github.com/davidfowl/SignalRCoreAlphaSample/blob/350420c3b075c02e2ae31431e0f698072fcfe721/Sample/wwwroot/index.html#L9

Doing those 2 things will let us do:

```js
let connection = new signalR.HubConnection('/chat');
```

**Comment by davidfowl** [comment]
We need this for the alpha.

**Comment by moozzyk** [comment]
This is basically a duplicate of https://github.com/aspnet/SignalR/issues/825 (which I closed by accident)

**Comment by moozzyk** [comment]
Fixed in bb79a9760c7052622bbc5aaeabb9fe7a600b9b84 and 67b2b27ac9573d6b453cef1a53efbcc22048c12e
